### DISCUSSION TITLE
renovate does not update package-lock.json

### DISCUSSION THREAD

**Discussion by ronaldploeger** [renovatebot/renovate#8694] [Request Help] [answered]
From the FAQ https://docs.renovatebot.com/faq/ I understand that the default behavior is: 

`Update yarn.lock and/or package-lock.json files if found`

and from the configuration docs https://docs.renovatebot.com/configuration-options/#updatelockfiles that the default value for "updateLockFiles" is true. 

But for some reason renovate does not update the package-lock.json but only the dependency version in package.json. 

This leads to all our build jobs failing as we run "npm ci" which complains: 

`cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.`

Are we doing something wrong? Missing a config?

**Comment by rarkins** [comment, marked as answer]
How you are running Renovate? (self hosted? which image?)

Are you debug logging? What does it say for the branch in question? Note: you must capture the log WHEN the commit is made - running on subsequent runs won't help.

**Reply by ronaldploeger** [reply]
We run renovate self-hosted (using the docker image renovate/renovate:24.46.0). I switched on debug logs and got: 

```
lock file error

Command failed: npm install --package-lock-only --ignore-scripts --no-audit
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for xxx@0.0.0: wanted: {"node":"12"} (current: {"node":"14.15.5","npm":"6.14.11"})
npm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0
npm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0
npm ERR! notsup Required: {"node":"12"}
npm ERR! notsup Actual:   {"npm":"6.14.11","node":"14.15.5"}
```

So the problem seems to be that the docker image runs node 14 and we require node 12. I guess we will upgrade to node 14. 

In general I think this should be logged as an error and not on level "debug".

**Reply by viceice** [reply]
You should think about using the renovate slim image with [binarySource=docker](https://docs.renovatebot.com/self-hosted-configuration/#binarysource) , this will solve your problem without forced node v14 migration. 😉 

`binarySource=docker` is preconfigureg for the lim image. you just need to mount the docker socket

it also supports dind if there are any concerns about reusing root docker host.

**Reply by rarkins** [reply]
It's intended that you receive that error in a comment in the PR too

**Reply by ronaldploeger** [reply]
Yes - you are right - it is in the PR. I just did not see it as I stopped at the failed build job. I still think it would be good to log this kind of behaviour on level "error" or "warn".

**Reply by rarkins** [reply]
it's already logged as a warning: https://github.com/renovatebot/renovate/blob/44004bf2d6904e510e374d0437d56689014aa1f6/lib/workers/branch/index.ts#L615-L619

**Reply by ronaldploeger** [reply]
@rarkins: For me this logs with level 20:

`{... "level":20, ... ,"msg":"lock file error",...}`

**Reply by ronaldploeger** [reply]
Hi @viceice, I am not quite sure how it works with the slim image and sidecar containers and I can not find any documentation. Can you point me to the docs or explain in more detail?

**Reply by rarkins** [reply]
> @rarkins: For me this logs with level 20:
> 
> `{... "level":20, ... ,"msg":"lock file error",...}`

Please provide the full log, where you see "lock file error" but not "artifactErrors". It should not be possible to see one without the other.

**Reply by viceice** [reply]
@ronaldploeger How are you running renovate currently? If it's  a simple docker container, you can continue this

```sh
docker run --rm \
  -v /tmp:/tmp \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v $PWD/config.js:/src/config.js \
  -w /src \
  -e LOG_LEVEL=debug \
  renovate/renovate:24.49.8-slim
```

**Reply by ronaldploeger** [reply]
Hi @rarkins,
this is the full log message: 

`{"name":"renovate","hostname":"renovate-manual-r85-wgz72","pid":8,"level":20,"logContext":"qlJgPYgvv","repository":"xxx/xxx","branch":"feature-renovate-husky-5.x","err":{"killed":false,"code":1,"signal":null,"cmd":"npm install --package-lock-only --ignore-scripts --no-audit","stdout":"","stderr":"npm ERR! code ENOTSUP\nnpm ERR! notsup Unsupported engine for xxx@0.0.0: wanted: {\"node\":\"12\"} (current: {\"node\":\"14.15.5\",\"npm\":\"6.14.11\"})\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Required: {\"node\":\"12\"}\nnpm ERR! notsup Actual:   {\"npm\":\"6.14.11\",\"node\":\"14.15.5\"}\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /tmp/renovate/cache/others/npm/_logs/2021-02-15T14_35_35_707Z-debug.log\n","message":"Command failed: npm install --package-lock-only --ignore-scripts --no-audit\nnpm ERR! code ENOTSUP\nnpm ERR! notsup Unsupported engine for xxx@0.0.0: wanted: {\"node\":\"12\"} (current: {\"node\":\"14.15.5\",\"npm\":\"6.14.11\"})\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Required: {\"node\":\"12\"}\nnpm ERR! notsup Actual:   {\"npm\":\"6.14.11\",\"node\":\"14.15.5\"}\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /tmp/renovate/cache/others/npm/_logs/2021-02-15T14_35_35_707Z-debug.log\n","stack":"Error: Command failed: npm install --package-lock-only --ignore-scripts --no-audit\nnpm ERR! code ENOTSUP\nnpm ERR! notsup Unsupported engine for xxx@0.0.0: wanted: {\"node\":\"12\"} (current: {\"node\":\"14.15.5\",\"npm\":\"6.14.11\"})\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Not compatible with your version of node/npm: xxx@0.0.0\nnpm ERR! notsup Required: {\"node\":\"12\"}\nnpm ERR! notsup Actual:   {\"npm\":\"6.14.11\",\"node\":\"14.15.5\"}\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /tmp/renovate/cache/others/npm/_logs/2021-02-15T14_35_35_707Z-debug.log\n\n    at ChildProcess.exithandler (child_process.js:308:12)\n    at ChildProcess.emit (events.js:315:20)\n    at ChildProcess.EventEmitter.emit (domain.js:467:12)\n    at maybeClose (internal/child_process.js:1048:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)"},"type":"npm","msg":"lock file error","time":"2021-02-15T14:35:35.717Z","v":0}`

**Reply by ronaldploeger** [reply]
@viceice We are running renovate as a k8s CronJob

**Reply by viceice** [reply]
The logs clearly says: image node version v 14 is not compatible with requested node v12

The best working set i know about renovate and kubernetes is that you need to start a docker dind pod with a shared volume ( `/tmp` by renovate default) and let renovate slim using that. eg something like:

```yaml
containers:
  - name: renovate
    image: renovate/renovate:24.11.2-slim
    env:
      - name: DOCKER_HOST
        value: 127.0.0.1
    volumeMounts:
      - name: renovate-tmp-volume
        mountPath: /tmp
  - name: renovate-dind
    image: docker:19.03.14-dind
    env:
    - name: DOCKER_TLS_CERTDIR
      value: ""
    securityContext:
      privileged: true
    volumeMounts:
    - name: renovate-tmp-volume
      mountPath: /tmp
volumes:
  - name: renovate-tmp-volume
    emptyDir: {}
```

ref #8169

**Reply by viceice** [reply]
renovatebot/helm-charts#91
